tests/modules/postnotifications.c

```diff
@@ -186,6 +190,32 @@ static int KeySpace_PostNotificationsAsyncSet(RedisModuleCtx *ctx, RedisModuleSt
     return REDISMODULE_OK;
 }
 
+static void KeySpace_ServerEventCallback(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent, void *data) {
+    REDISMODULE_NOT_USED(eid);
+    REDISMODULE_NOT_USED(data);
+    if (subevent > 3) {
+        RedisModule_Log(ctx, "warning", "Got an unexpected subevent '%ld'", subevent);
+        return;
+    }
+    static const char* events[] = {
+            "before_deleted",
+            "before_expired",
+            "before_evicted",
+            "before_overwritten",
+    };
+
+    const RedisModuleString *key_name = RedisModule_GetKeyNameFromModuleKey(((RedisModuleKeyInfo*)data)->key);
+    const char *key_str = RedisModule_StringPtrLen(key_name, NULL);
+    const char *event = events[subevent];
+    size_t event_len = strlen(event);
+    if (strncmp(key_str, event , event_len) == 0) {
+        return; /* avoid loops */
+    }
```

Comment:
don't we want to skip any event on any key named after an event?
i.e. not just skip "evicted" event on the "evicted" key, but also skip "deleted" on the "evicted" key?

Reply:
Yes but this is the only event that might happened on our test flow, we never explicitly delete or expire the deleted/expired keys. I can either add what you are suggesting or change it to be specifically on evicted, I think the first is better, WDYT?

Reply:
yes, the first option would be better.
another alternative which i would be ok with, is just add a comment saying what you just said.
i.e. that we generally don't want to log any event on our tracking keys, but the only one that actually happens in the current code is "evicted" event on the "evicted" key.

p.s. why won't we have "evicted" event on the "expired" key?

Reply:
> p.s. why won't we have "evicted" event on the "expired" key?

Just because we do not have a test that does it (configure eviction and create keys with expiration). I will change the code according to your suggestion.